keanu-project/src/main/java/io/improbable/keanu/vertices/bool/probabilistic/ProbabilisticBooleanVertex.java

```diff
@@ -0,0 +1,24 @@
+package io.improbable.keanu.vertices.bool.probabilistic;
+
+import io.improbable.keanu.tensor.TensorShapeValidation;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.vertices.bool.BoolVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+
+public abstract class ProbabilisticBooleanVertex extends BoolVertex implements ProbabilisticBoolean {
```

Comment:
I am not keen on introducing more abstract classes into the hierarchy. Since `sample` is a member of `Vertex`, that seems like the natural place for these new methods too.